xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -589,7 +591,12 @@ bool CGUIDialogPVRTimerSettings::Save()
     }
     else if (m_endLocalTime < m_startLocalTime) // Assume the user knows what they are doing, but log a warning just in case
     {
-      CLog::Log(LOGWARNING, "Timer settings dialog: Specified recording end time < start time: expect errors!");
+      CLog::Log(LOGWARNING,
+                "Timer settings dialog: Specified end time must be greater than start time: "
+                "please fix before adding/updating a timer");
+      HELPERS::ShowOKDialogText(CVariant{19065}, // "Timer settings"
+                                CVariant{19072}); // In order to add/update a timer
+      bValid = false;
     }
     m_timerInfoTag->SetStartFromLocalTime(m_startLocalTime);
```

Comment:
Is it intended that you already update `m_timerInfoTag` even though the input going into it has been considered invalid? Can't you just completely abort here and only execute the rest of the code if the input is valid?

Reply:
Good point.